### ISSUE TITLE
Version 51.10 Broke Package.xml Deploys & Retrieves

### ISSUE THREAD

**Issue by csosa27** [forcedotcom/salesforcedx-vscode#3179]
### Summary
After upgrading to the 51.10 version, I was not able to deploy or retrieve components via the manifest > Package.xml. 

I had to install an older version for it to work. Installed 51.7, and it did the trick.

### Steps To Reproduce:

1.  Insert a component Type> member and name tags to you package.xml file within the Package tag. Like the following
```
<types>
  <members>[Enter component name here]</members>
  <name>CustomMetadata</name>
</types>
```
2. Right click on the package.xml file and select retrieve or deploy.
3. It will try to run, but will ultimately fail.

### Expected result

It should retrieve or deploy said component files

### Actual result

Received an Unknown Exception on the output. 

### Additional information

_Feel free to attach a screenshot_.

**VS Code Version**: 1.55.2 Version: 1.55.2 (user setup)
Date: 2021-04-13T09:35:57.887Z
Electron: 11.3.0
Chrome: 87.0.4280.141
Node.js: 12.18.3
V8: 8.7.220.31-electron.0

**SFDX CLI Version**:  v51.7.0

**OS and version**: Windows_NT x64 10.0.18363

**Comment by sfsholden** [comment]
@csosa27 Thank you for reaching out to us. 

In release 51.8.0 we switched the `Experimental: Deploy Retrieve` setting to on by default (mentioned [here](https://github.com/forcedotcom/salesforcedx-vscode/issues/3109)). This new route using the Source Deploy Retrieve Library is much faster than the CLI route. However, we're still working out some kinks. 

Workaround: Disable `Experimental: Deploy Retrieve`

We do have a change coming in #3178 as part of release 51.11.0 (which should hopefully be published today). This change should help determine the root cause for why this issue is happening.

**Comment by berghoc** [comment]
Ok thank god, I'm not crazy. Just started working with Metadata in VSC for the first time, all the guides mention these commands for the package.xml and they weren't there for me. Couldn't find anything online to fix this, I guess its because its a new issue. Will try the workaround for now.
![package](https://user-images.githubusercontent.com/25653082/116621242-e1ee6500-a910-11eb-8060-26cbbefadd2f.png)

**Comment by berghoc** [comment]
> Ok thank god, I'm not crazy. Just started working with Metadata in VSC for the first time, all the guides mention these commands for the package.xml and they weren't there for me. Couldn't find anything online to fix this, I guess its because its a new issue. Will try the workaround for now.
> ![package](https://user-images.githubusercontent.com/25653082/116621242-e1ee6500-a910-11eb-8060-26cbbefadd2f.png)

Disabled the experimental setting and still no Deploy/Retrieve options :(

**Comment by sfsholden** [comment]
Hi @berghoc, that sounds like an unrelated issue.

Can you confirm for me - are you authenticated to an org? And if yes, what type of org?

Deploy/Retrieve commands are not available when working within a scratch org. [This](https://developer.salesforce.com/tools/vscode/en/user-guide/development-models) is a great source if you'd like to learn more.

**Comment by sfsholden** [comment]
@csosa27 VSCode Extensions v51.11.0 has been published. 

Please [enable the setting](https://developer.salesforce.com/tools/vscode/ja/user-guide/perf-enhancements/#setup) `Experimental: Deploy Retrieve` and attempt your retrieve/deploy again. This likely isn't going to solve your issue, but it should provide a more descriptive error that will help us determine the root issue. We are very interested in knowing what error you see!

Afterwards, to unblock the retrieve/deploy you can disable the `Experimental: Deploy Retrieve` setting. 

Thank you again for your help in getting to the root of this problem!

**Comment by Techley-git** [comment]
I think I may be having a similar issue. 
When I try to right click and retrieve source from org on my package XML I get
16:22:09.369 Starting SFDX: Retrieve Source from Org


16:30:27.606 ended SFDX: Retrieve Source from Org
When I run the command I get

PS C:\Users\user\SFDX VSC\EOS1Backup-05-03-21> sfdx update
sfdx-cli: Updating CLI... already on latest version: 7.99.0-79e53c6
PS C:\Users\user\SFDX VSC\EOS1Backup-05-03-21> sfdx force:source:retrieve -x manifest\packageALL.xml
ERROR running force:source:retrieve:  The specified metadata type is unsupported: [Audience]

![image](https://user-images.githubusercontent.com/72884210/116949318-afa57600-ac36-11eb-9715-6afaeabd3b22.png)

**Comment by sfsholden** [comment]
@Techley-git Could you confirm - which version of the extensions are you currently using?

Also, do you see any additional output in the Developer Console?
From VSCode > Bring up the `Command Palette` > Search for `Developer: Toggle Developer Tools` > If there is further information, it should be under the `Console` tab.

**Comment by Techley-git** [comment]
I'm guessing something may have updated because I get a different experience today. Might be unrelated to this thread.

Console:
```
Extension Host
localProcessExtensionHost.ts:273 Debugger attached.
log.ts:290  WARN UNRESPONSIVE extension host, 'salesforce.salesforcedx-vscode-core' took 99% of 4018.906ms, saved PROFILE here: 'file:///c%3A/Users/userAppData/Local/Temp/exthost-bdab5f.cpuprofile' (6) [{…}, {…}, {…}, {…}, {…}, {…}]
notificationsAlerts.ts:40 SFDX: Retrieve Source from Org failed to run
onDidChangeNotification @ notificationsAlerts.ts:40
(anonymous) @ notificationsAlerts.ts:26
fire @ event.ts:622
addNotification @ notifications.ts:206
notify @ notificationService.ts:137
(anonymous) @ mainThreadMessageService.ts:86
_showMessage @ mainThreadMessageService.ts:44
$showMessage @ mainThreadMessageService.ts:38
_doInvokeHandler @ rpcProtocol.ts:417
_invokeHandler @ rpcProtocol.ts:402
_receiveRequest @ rpcProtocol.ts:318
_receiveOneMessage @ rpcProtocol.ts:245
(anonymous) @ rpcProtocol.ts:110
fire @ event.ts:622
fire @ ipc.net.ts:468
_receiveMessage @ ipc.net.ts:821
(anonymous) @ ipc.net.ts:660
fire @ event.ts:622
acceptChunk @ ipc.net.ts:241
(anonymous) @ ipc.net.ts:202
k @ ipc.net.ts:50
emit @ events.js:315
addChunk @ _stream_readable.js:295
readableAddChunk @ _stream_readable.js:271
Readable.push @ _stream_readable.js:212
onStreamRead @ internal/stream_base_commons.js:186
```
![image](https://user-images.githubusercontent.com/72884210/117095286-d2a95600-ad1a-11eb-8804-94c0ff29bec7.png)
Output: 
![image](https://user-images.githubusercontent.com/72884210/117095293-d9d06400-ad1a-11eb-819d-6c17e904198e.png)

**Comment by sfsholden** [comment]
@Techley-git Yes, that does look like a separate issue. Please open a new issue with the information from above. i.e. the error for `Could not infer a metadata type` in regards to reports.

**Comment by sfsholden** [comment]
Closing issue. 

The original problem due to Unknown Exception was resolved with #3178. This change was deployed as part of v51.11.0.

Note that this change will help us determine the real root of issues that are occurring with the new deploy/retrieve route. Please continue to open issues as this reveals gaps in coverage we need to add!

Thank you!
